src/internal/controllers/admins_controller.go

```diff
@@ -4,73 +4,137 @@ import (
 	"birdai/src/internal/models"
 	"birdai/src/internal/utils"
 	"net/http"
+
+	"go.mongodb.org/mongo-driver/bson"
 )
 
 func (c *Controller) CGetAdminById(id string) (models.Response) {
-	/*
-	coll := c.db.GetCollection(repositories.AdminColl)
-	filter := bson.M{"_id": id}
-	response := coll.FindOne(filter)
-	return response
-	*/
-	return utils.ErrorNotImplemented("CGetAdminById")
+	response := c.db.Admin.GetAdminById(id)
+	if utils.IsTypeError(response) {
+		return response
+	}
+	admin := response.Data.(*models.AdminDB)
+	adminResponse := c.AdminDBToOutput(admin)
+	return adminResponse
 }
 
-func (c *Controller) CListAdmins(set int, search string) (models.Response) {
-	/*
-	coll := c.db.GetCollection(repositories.AdminColl)
-	response := coll.FindAll()
-	return response
-	*/
-	return utils.ErrorNotImplemented("CListAdmins")
+func (c *Controller) CGetAdminByUserId(id string) (models.Response) {
+	response := c.db.Admin.GetAdminByUserId(id)
+	if utils.IsTypeError(response) {
+		return response
+	}
+	admin := response.Data.(*models.AdminDB)
+	adminResponse := c.AdminDBToOutput(admin)
+	return adminResponse
 }
 
-func (c *Controller) CCreateAdmin(userId string, admin *models.AdminInput) (models.Response) {
-	/*
-	coll := c.db.GetCollection(repositories.AdminColl)
-	response := coll.CreateOne(admin)
-	return response
-	*/
-	return utils.ErrorNotImplemented("CCreateAdmin")
+func (c *Controller) CListAdmins(set int, search string) (models.Response) {
+	response := c.db.Admin.ListAdmins(bson.M{}, set)
+	if utils.IsTypeError(response) {
+		return response
+	}
+
+	output := []*models.AdminOutput{}
+	for _, admin := range response.Data.([]models.AdminDB) {
+		adminResponse := c.AdminDBToOutput(&admin)
+		if utils.IsTypeError(adminResponse) {
+			return adminResponse
+		}
+		
+		output = append(output, adminResponse.Data.(*models.AdminOutput))
+	}
+
+	return utils.Response(output)
 }
 
-func (c *Controller) CUpdateAdmin(userId string, admin *models.AdminInput) (models.Response) {
-	//check if updating last superadmin to admin
+func (c *Controller) CCreateAdmin(adminInput *models.AdminInput) (models.Response) {
+	currentAdmin := c.db.Admin.GetAdminByUserId(adminInput.UserId)
+	if utils.IsTypeError(currentAdmin) && currentAdmin.Data.(models.Err).StatusCode != http.StatusNotFound{
+		return currentAdmin
+	}
+	if utils.IsType(currentAdmin, models.AdminOutput{}) {
+		return utils.ErrorToResponse(http.StatusConflict, "Admin already exists", "Admin with that user id already exists")
+	}
```

Comment:
Just nu går det att gör samma person till admin flera gånger. Skulle tro att det är checken som inte blir rätt gjord för stunden. Ska sätta mig och testa göra om den också

![image](https://github.com/TovahParnes/BirdAI_D7057E/assets/60612941/91ae973d-663c-47b1-8b0e-b7bb8b2b6097)

Reply:
Jag uppdaterade checken i CCreateAdmin till att kolla efter *models.adminDB{} istället och nu gav den error 
```suggestion
func (c *Controller) CCreateAdmin(adminInput *models.AdminInput) (models.Response) {
	currentAdmin := c.db.Admin.GetAdminByUserId(adminInput.UserId)
	if utils.IsTypeError(currentAdmin) && currentAdmin.Data.(models.Err).StatusCode != http.StatusNotFound{
		return currentAdmin
	}
	if utils.IsType(currentAdmin, &models.AdminDB{}) {
		return utils.ErrorToResponse(http.StatusConflict, "Admin already exists", "Admin with that user id already exists")
	}
```
Då fick jag detta istället
![image](https://github.com/TovahParnes/BirdAI_D7057E/assets/60612941/4dc74039-e62b-43c1-9703-8b3f3584e55e)